src/components/Table/GridTable.tsx

```diff
@@ -401,7 +401,9 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     visibleDataRows = visibleDataRows.slice(0, filterMaxRows + keptSelectedRows.length);
   }
 
-  tableState.setMatchedRows(filteredRowIds);
+  useEffect(() => {
+    tableState.setMatchedRows(filteredRowIds);
+  }, [tableState, filteredRowIds]);
```

Comment:
Mutating observables directly from render causes those "Cannot update" warnings, so if we do it from a `useEffect`, it'll be fine/albeit might delay an update:

https://github.com/facebook/react/issues/26962#issuecomment-1656742586